### PR TITLE
Fix calicoctl update and usage

### PR DESCRIPTION
**Pull request by tomastigera**
- clean must remove dist/calicoctl to force updating it

- calicoctl version reports verion used properly

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,6 +160,7 @@ clean:
 	find . -name '*.created' -exec rm -f {} +
 	find . -name '*.pyc' -exec rm -f {} +
 	rm -rf certs *.tar vendor $(NODE_CONTAINER_BIN_DIR)
+	rm dist/calicoctl
```
Comment (jpalardy) [inline Makefile]:
```suggestion
	rm -f dist/calicoctl
```

without `-f`, the exit code of rm != 0 if the file is missing.

Should the whole `dist` directory be deleted?

Comment (tomastigera) [reply Makefile]: - or -f ... ack

I would guess dist/*, waiting for opinion from @caseydavenport & co.

Comment (caseydavenport) [reply Makefile]: Yeah I think we can blow away all of `dist` in `clean`

Comment (tomastigera) [reply Makefile]: ack, done

**Pull request merged**